api/types.proto

```diff
@@ -506,6 +506,21 @@ message PortConfig {
 		UDP = 1 [(gogoproto.enumvalue_customname) = "ProtocolUDP"];
 	}
 
+	enum Exposition {
+		option (gogoproto.enum_customname) = "PortConfigExposition";
+		option (gogoproto.goproto_enum_prefix) = false;
+
+		// PortConfigExpositionIngress exposes the port across the cluster on
+		// all nodes.
+		INGRESS = 0;
+
+		// PortConfigExpositionHost exposes the port on just the target host.
+		// If the published port is undefined, an ephemeral port will be
+		// allocated. If the published port is defined, the node will attempt
```

Comment:
@mavenugo If the user defines a publish port in host mode, only one instance of the task can run per node. If the user leaves this undefined, each instance will take a new ephemeral port.